### ISSUE TITLE
How to get the real IE Version?

### ISSUE THREAD

**Issue by kermit-the-frog** [HaraldWalker/user-agent-utils#83]
Hi,
IE 11 running in compatibility mode has the following user agent string:

    Mozilla/4.0 (compatible; MSIE 7.0; Windows NT 6.1; WOW64; Trident/7.0; SLCC2; .NET CLR 2.0.50727; .NET CLR 3.5.30729; .NET CLR 3.0.30729; Media Center PC 6.0; .NET4.0C; .NET4.0E)

IE 9 running in compatibility mode has the following user agent string:
    Mozilla/4.0 (compatible; MSIE 7.0; Windows NT 6.1; WOW64; Trident/5.0; SLCC2; .NET CLR 2.0.50727; .NET CLR 3.5.30729; .NET CLR 3.0.30729; Media Center PC 6.0)

For both `eu.bitwalker.useragentutils.UserAgent.getBrowser` returns `eu.bitwalker.useragentutils.Browser.IE7, because there is a `MSIE 7.0` in the strings. Is there any way to also get the real browser version given by the `Trident/7.0` or `Trident/5.0`?

**Comment by jeffreytgilbert** [comment]
Check it on the browser side. User agents can be false and commonly are with IE.

**Comment by jomu75** [comment]
Hi,
I agree, user agent strings (especially IE) are not very reliable. But we have to identify the browser version on the server side.

We encounter the same problem with a slightly different UserAgent:
<pre><code>Mozilla/4.0 (compatible; MSIE 7.0; Windows NT 6.1; WOW64; Trident/6.0; SLCC2; .NET CLR 2.0.50727; .NET CLR 3.5.30729; .NET CLR 3.0.30729; Media Center PC 6.0; .NET4.0C; .NET4.0E)</code></pre>

Method "userAgent.getBrowser()" returns IE7

We started to implement some workarounds but it would be great if a solution can be provided directly here.

**Comment by kermit-the-frog** [comment]
@jomu75
`Trident/6.0` is IE 10.0 isn't it?

Did you already implement a working workaround, that could be shared?

**Comment by jomu75** [comment]
Yes, correct - Trident/6.0 is IE10. But the problem is the same that the evaluation returns IE7.

Just played around and we do not have a final solution. We modified the following line in class <code>Browser.java</code>:
<pre><code>IE10(Manufacturer.MICROSOFT, Browser.IE, 92, "Internet Explorer 10", new String[] { "Trident/6.0", "MSIE 10" }, null, BrowserType.WEB_BROWSER, RenderingEngine.TRIDENT, null ),   // before MSIE</code></pre>
This returns "IE10" when calling <code>getBrowser()</code> but calling <code>getBrowserVersion()</code> still returns "7.0"

**Comment by HaraldWalker** [comment]
What is the use case? I think if newer IE versions have been set to behave like an ancient IE 7, then it is correct to return IE7, for instance when you use this to provide a certain html version.
Would be nice to know maybe that it is a newer browser running in compatibility mode but since this is as far as I know the only browser behaving like this, it might be better if you just handle it separately. If IE7, then check user-agent and do something with it. Since the market share of IE7 is dropping quickly that shouldn't happen too often.

**Comment by HaraldWalker** [comment]
Maybe the IE7 regex for the version can be changed in a way that it returns the correct version if a newer IE is running in compatibility mode.

**Comment by kermit-the-frog** [comment]
That sounds great 
let `eu.bitwalker.useragentutils.UserAgent.getBrowser` return `eu.bitwalker.useragentutils.Browser.IE7` and let `eu.bitwalker.useragentutils.UserAgent.getBrowserVersion`return the correct version.

**Comment by kermit-the-frog** [comment]
A new idea came in my mind: what about defining a new browser `eu.bitwalker.useragentutils.Browser.IE7Compat` with it's own regex.

**Comment by HaraldWalker** [comment]
Then those sites using an IE7 browser switch will not provide the correct version unless they and IE7Compat to their code.

**Comment by kermit-the-frog** [comment]
Ok, I see. But changing the version would break with #31.

So we need something different, for excample `eu.bitwalker.useragentutils.UserAgent.getBrowserVersion(boolean showRealIEVersion)` and a second `versionRegEx`
